app/javascript/components/async-credentials/async-credentials.jsx

```diff
@@ -75,7 +82,9 @@ const AsyncCredentials = ({
 
         <FormSpy subscription={{ values: true }}>
           {() => {
-            const depsValid = validateDependentFields();
+            // The list of registered fields shows up after this render, so the validation has to happen
+            // with a delay and has to be pulled back via the state.
+            setTimeout(() => setState(state => ({ ...state, depsValid: validateDependentFields() })));
```

Comment:
@skateman Have you try using a component with `useEffect`? Making any `setState` calls during render can cause infinite loops.

Reply:
I was thinking about it, but I don't know what should I put into the array in `useEffect`...

Reply:
Nothing

Reply:
:eyes:

Reply:
@Hyperkid123 this doesn't even get called upon a schema condition change :confused: 
```js
  useEffect(() => {
    console.log('called');
    validateDependentFields();
  });
```

Reply:
Right it's not a prop or state change :facepalm: the form spy reacts only on values changes. You can ignore the comment.

Reply:
Well, I think we shouldn't rely on setTimeout, all you have to do to break it is add another layer of async.
So, sounds like an architectural problem, shouldn't the FormSpy get called on all such changes? Or do we need `debug` again?

But I'm :+1: with this if it works at least now :).

Reply:
The form spy is subscribed only to value changes at this point. So the state does not have to be in sync with the form state.

Reply:
So, the solution is to subscribe to ..something else?

Reply:
The solution is to remove the subscription limitation (don't have the prop). But that will come with a performance hit since all children will be re-renderer od each form state change. It might be acceptable to you. I don't know what will be the real impact. But then it may cause the issue by setting a state inside the render cycle.

The setTimeout is, unfortunately, an "official" suggestion of the final from the library developer for cases like this. We are moving to a custom form state management but that will take some time until its finished.